sdk/openai/openai/src/api/oaiSse.ts

```diff
@@ -10,29 +10,35 @@
  */
 
 import { StreamableMethod } from "@azure-rest/core-client";
-import { getSSEs } from "./getSSEs.js";
+import { getStream } from "./getSSEs.js";
 import { wrapError } from "./util.js";
+import { EventStream } from "../models/models.js";
+import { EventMessage, createSseStream } from "@azure/core-sse";
+import { polyfillStream } from "./readableStreamUtils.js";
 
-export async function* getOaiSSEs<TEvent, O extends Record<string, any>>(
+export async function getOaiSSEs<TEvent, O extends Record<string, any>>(
   response: StreamableMethod<unknown>,
   toEvent: (obj: O) => TEvent
-): AsyncIterable<TEvent> {
-  const stream = await getSSEs(response);
-  let isDone = false;
-  for await (const event of stream) {
-    if (isDone) {
-      // handle a case where the service sends excess stream
-      // data after the [DONE] event
-      continue;
-    } else if (event.data === "[DONE]") {
-      isDone = true;
-    } else {
-      yield toEvent(
-        wrapError(
-          () => JSON.parse(event.data),
-          "Error parsing an event. See 'cause' for more details"
+): Promise<EventStream<TEvent>> {
+  const stringStream = await getStream(response);
+  // TODO: use await using once we drop support for NodeJS v18.
```

Comment:
Just curious, what would using `await using` look like here? Would it even be possible, since we aren't actually doing the disposal before we return from the function?

If it is possible, from the [release notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-2.html#using-declarations-and-explicit-resource-management) it sounded to me like `await using` should transpile down and work properly provided we polyfill `Symbol.asyncDispose`, which we are now doing.

Reply:
You're right, disposition is strictly tied to the static scope:

> When a using declaration is parsed with BindingIdentifier Initializer, the bindings created in the declaration are tracked for disposal at the end of the containing Block or Module (a using declaration cannot be used at the top level of a Script)